src/models/assistant-model.ts

```diff
@@ -466,79 +471,78 @@ export const AssistantModel = types
-    //     }
-    //     self.isCancelling = false;
-    //   }
-    // });
-
-    // const resetThread = flow(function* () {
-    //   try {
-    //     self.isResetting = true;
-    //     const threadId = self.thread.id;
-    //     const allThreadMessages = yield self.apiConnection.beta.threads.messages.list(threadId);
-    //     yield deleteThread();
-    //     yield createThread();
-    //     yield fetchAndSendDataContexts();
-    //     self.addDbgMsg("Sending thread history to LLM", formatJsonMessage(allThreadMessages));
-    //     yield self.apiConnection.beta.threads.messages.create(self.thread.id, {
-    //       role: "user",
-    //       content: `This is a system message containing the previous conversation history. ${allThreadMessages}`,
-    //     });
-    //     self.isResetting = false;
-    //   } catch (err) {
-    //     console.error("Failed to reset thread:", err);
-    //     self.addDbgMsg("Failed to reset thread", formatJsonMessage(err));
-    //     self.isCancelling = false;
-    //     self.isResetting = false;
-    //   }
-    // });
+    const cancelRun = flow(function* (runId: string) {
+      try {
+        const reqBody = { messageId: runId };
+        const response = yield postMessage(reqBody, "cancel");
 
-    // const createThread = flow(function* () {
-    //   try {
-    //     const newThread = yield self.apiConnection.beta.threads.create();
-    //     self.thread = newThread;
-    //   } catch (err) {
-    //     console.error("Error creating thread:", err);
-    //   }
-    // });
+        if (!response.ok) {
+          throw new Error(`Failed to cancel run: ${response.statusText}`);
+        }
 
-    // const deleteThread = flow(function* () {
-    //   try {
-    //     if (!self.thread) {
-    //       console.warn("No thread to delete.");
-    //       return;
-    //     }
+        const cancelRes = yield response.json();
+        self.isCancelling = false;
+        self.setShowLoadingIndicator(false);
+        self.addDbgMsg(`Cancel request received`, formatJsonMessage(cancelRes));
+      } catch (err: any) {
+        const errorMessage =
+          err instanceof Error
+            ? err.message
+            : typeof err === "string"
+            ? err
+            : JSON.stringify(err);
+        if (errorMessage.includes("Message not found or already completed")) {
+          self.isCancelling = false;
+          return;
+        } else {
+          console.error("Failed to cancel run:", errorMessage);
+          self.addDbgMsg("Failed to cancel run", formatJsonMessage(errorMessage));
+        }
+        self.isCancelling = false;
+      }
+    });
 
-    //     const threadId = self.thread.id;
-    //     yield requestThreadDeletion(threadId);
-    //     self.addDbgMsg("Thread deleted", `Thread with ID ${threadId} deleted`);
-    //     self.thread = undefined;
+    const resetThread = flow(function* () {
```

Comment:
Does this need to cancel any in-progress request?

Reply:
Yeah, it should. Thanks for catching that. Although, it seems like we're no longer using `resetThread`. I've commented the function out (will remove if we truly have no need for it at the end of the LangChain-related changes), but I did just now add in-progress cancellation to the `createThread` function which does still get used.